kv/range_cache.go

```diff
@@ -149,7 +162,10 @@ func (rmc *RangeDescriptorCache) EvictCachedRangeDescriptor(key proto.Key) {
 // the cache.
 func (rmc *RangeDescriptorCache) getCachedRangeDescriptor(key proto.Key) (
 	rangeCacheKey, *proto.RangeDescriptor) {
-	metaKey := engine.RangeMetaKey(key)
+	// We want to look up the range descriptor for key. The cache is
+	// indexed using the end-key of the range, but the end-key is
+	// non-inclusive. So we access the cache using key.Next().
```

Comment:
I can't see the reason why we use EndKey and Ceil instead of StartKey and Floor. I'm sure it exists and will add a comment when I discover it.